src/templates/cmp.js

```diff
@@ -0,0 +1,187 @@
+    }
+
+    function log(event, success, parameters) {
+      window.__tcfapi('log', 2, function () {}, btoa(JSON.stringify({ event: event, parameters: parameters })));
+    }
+
+    function isIframeCapable() {
+      return (
+        window.navigator &&
+        navigator.userAgent &&
+        navigator.userAgent.indexOf &&
+        navigator.userAgent.indexOf('Presto') === -1
+      );
+    }
+
+    function createIframe() {
+      var iframe = document.createElement('iframe');
+
+      iframe.setAttribute(
+        'src',
+        '<%-URL_SCHEME%>://<%-CONSENT_SERVER_HOST%>/<%-API_VERSION%>/iframe.html' +
+          (channelId !== '' ? '?channelId=' + channelId : ''),
+      );
+      iframe.setAttribute('style', 'border:0;outline:0;width:0;height:0;');
+      iframe.setAttribute('frameborder', '0');
+
+      iframe.addEventListener('load', function () {
+        window.__tcfapi = function (command, version, callback, parameter) {
+          message('cmd;' + command + ';' + version + ';' + parameter, callback);
+        };
+
+        window.addEventListener(
+          'message',
+          function (ev) {
+            try {
+              if (ev.origin === '<%-URL_SCHEME%>://<%-CONSENT_SERVER_HOST%>' && ev.data) {
+                var message = ev.data.split(';');
+                var position = message[0] === 'err' ? 1 : 0;
+                var id = message[position];
+                var callback = callbackMap[id][position];
+                if (logCallbackIndex + '' !== id) delete callbackMap[id];
+                var r = JSON.parse(atob(message[++position]));
+                var s = message[++position] ? true : undefined;
+                if (callback) {
+                  callback(r, s);
+                }
+              }
+            } catch (e) {}
+          },
+          false,
+        );
+
+        onAPILoaded('iframe');
+      });
+
+      iframe.addEventListener('error', log.bind(null, 'loaded', false, { type: 'iframe' }));
+
+      return iframe;
+    }
+
+    function loadIframe(retriesLeft) {
+      if (retriesLeft < 0) {
+        return;
+      }
+
+      var body = document.getElementsByTagName('body')[0];
+      if (!body) {
+        setTimeout(loadIframe.bind(this, retriesLeft - 1), 100);
+        return;
+      }
+
+      iframe = createIframe();
+      body.appendChild(iframe);
+    }
+
+    function createTcfapiScriptTag() {
+      var xt = '<%-XT%>';
+      var hasConsent;
+
+      if (window.localStorage && localStorage.getItem && localStorage.setItem) {
+        if (!localStorage.getItem('<%-TECH_COOKIE_NAME%>')) {
+          // no tech info yet, need to store
+          localStorage.setItem('<%-TECH_COOKIE_NAME%>', xt);
+        } else {
+          xt = localStorage.getItem('<%-TECH_COOKIE_NAME%>'); // prefer tech info from localStorage
+          hasConsent = localStorage.getItem('<%-COOKIE_NAME%>');
+        }
+      }
```

Comment:
Would be great if we could extract this logic for getting the existing consent decision into a shared function .. as this is also needed in tcfapi.js